# Google Play Store App Success Analysis


## Problem Statement

Hence Google Play Store is consist of several of thousands application of different categories. Our job is to find the factor leading the app to success, factors could be behalf of Installs, on bhealf of Review & Ratings or behalf of Paid/Unpaid.

## Objective
- Identify factors leading to success
- Compare Unpaid vs Paid Apps
- Analyze impact of size

## Approach
- Data Gathering
- Data Cleaning & Transformation
- EDA
- Visualization
- Insights & Recommendations

## Data Gathering

### Data Set Location

In [ ]:
PlayStoreData = "/content/drive/MyDrive/Almabetter/Projects/PlayStore-AppReview/Dataset/Play Store Data.csv"
UserReviews = "/content/drive/MyDrive/Almabetter/Projects/PlayStore-AppReview/Dataset/User Reviews.csv"

### Basic Imports

In [ ]:
# Data Handling
import pandas as pd
import numpy as np

# Visualization
import plotly.express as px

In [ ]:
# For rendering plotly graphs in Notebook
import plotly.io as pio
pio.renderers.default = "colab"

### Dataframes

In [ ]:
playstore = pd.read_csv(PlayStoreData)
userreviews = pd.read_csv(UserReviews)

In [ ]:
playstore.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up


In [ ]:
userreviews.head()

,App,Translated_Review,Sentiment,Sentiment_Polarity,Sentiment_Subjectivity
0,10 Best Foods for You,I like eat delicious food. That's I'm cooking ...,Positive,1.00,0.533333
1,10 Best Foods for You,This help eating healthy exercise regular basis,Positive,0.25,0.288462
2,10 Best Foods for You,NaN,NaN,NaN,NaN
3,10 Best Foods for You,Works great especially going grocery store,Positive,0.40,0.875000
4,10 Best Foods for You,Best idea us,Positive,1.00,0.300000


### About Data

In [ ]:
playstore.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [ ]:
userreviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


### Finding Nulls

In [ ]:
playstore.isnull().sum()

,0
App,0
Category,0
Rating,1474
Reviews,0
Size,0
Installs,0
Type,1
Price,0
Content Rating,1
Genres,0


In [ ]:
userreviews.isnull().sum()

,0
App,0
Translated_Review,26868
Sentiment,26863
Sentiment_Polarity,26863
Sentiment_Subjectivity,26863


### Finding Duplicates

In [ ]:
playstore.duplicated().sum()

np.int64(483)

In [ ]:
userreviews.duplicated().sum()

np.int64(33616)

## Data Exploration Summary

- Play Store dataset contains `10841` applications (rows) with `13` features.
- Application Review dataset is consist of `64295` reviews with `5` features.

### Key Issues:
- Several important columns has inappropriate types and needs type converstion.
- Rating column has `1474` missing values.
- User reviews has large number of columns with missing vlues(Translated_review, Sentiment, Sentiment_Polarity & Sentiment_Subjectivity)
- Having significantly large number of duplicate entries in **User Review Datasets** `33616` and a decent amount in **PlayStore Dataset** `483`

### Conclusion
- Data cleaning and Transformation is required.

## Cleaning

In [ ]:
playstore = playstore.drop_duplicates()
userreviews = userreviews.drop_duplicates()

## Transformation

In [ ]:
playstore['Installs'].value_counts()

,count
Installs,
"1,000,000+",1488
"10,000,000+",1132
"100,000+",1129
"10,000+",1033
"1,000+",890
100+,710
"5,000,000+",683
"500,000+",517
"50,000+",474


In [ ]:
playstore['Installs'] = playstore['Installs'].str.replace("Free", "0").str.replace("+", "", regex=False).str.replace(",", "").astype("float64")

In [ ]:
playstore['Installs'].value_counts()

,count
Installs,
1.000000e+06,1488
1.000000e+07,1132
1.000000e+05,1129
1.000000e+04,1033
1.000000e+03,890
1.000000e+02,710
5.000000e+06,683
5.000000e+05,517
5.000000e+04,474


In [ ]:
playstore['Price'].value_counts()

,count
Price,
0,9592
$0.99,146
$2.99,125
$1.99,73
$4.99,70
...,...
$394.99,1
$1.26,1
Everyone,1


In [ ]:
playstore['Price'] = playstore['Price'].str.replace("Everyone","0").str.replace("$", "").astype(np.float64)

In [ ]:
playstore['Price'] = round(playstore['Price']*94.33, 2)

In [ ]:
playstore['Price'].value_counts()

,count
Price,
0.00,9593
93.39,146
282.05,125
187.72,73
470.71,70
...,...
340.53,1
37259.41,1
118.86,1


In [ ]:
playstore['Reviews'].value_counts()

,count
Reviews,
0,593
1,272
2,213
3,170
4,137
...,...
240475,1
5282558,1
654418,1


In [ ]:
def reviewCount(review):
  try:
    return np.float64(review.replace(",", ""))
  except:
    return np.float64(review.replace("M", ""))*1000000
playstore['Reviews'] = playstore['Reviews'].apply(reviewCount)

In [ ]:
playstore['Size'].value_counts()

,count
Size,
Varies with device,1526
11M,188
13M,186
12M,186
14M,182
...,...
552k,1
885k,1
1020k,1


In [ ]:
def convertSize(size):
  if 'M' in size:
    return np.float64(size.replace("M", ""))
  elif 'k' in size:
    return np.float64(size.replace('k', ""))/1024
  elif 'G' in size:
    return np.float64(size.replace('G', ""))*1000
  else:
    return np.nan
playstore['Size'] = playstore['Size'].apply(convertSize)
playstore['Size'].fillna(playstore['Size'].median(), inplace=True)

/tmp/ipykernel_147670/590676104.py:11: FutureWarning:

A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.





In [ ]:
playstore['Size']

,Size
0,19.0
1,14.0
2,8.7
3,25.0
4,2.8
...,...
10836,53.0
10837,3.6
10838,9.5
10839,13.0


## Handling Missing and Droping Irrelavent

In [ ]:
# Type - removing the missing rows
# Content Rating - removing the missing rows
# Current Ver / Android Ver - application success doesn't matter on basis of version of application
playstore.dropna(subset=["Type", "Content Rating"], inplace=True)
playstore.drop(columns=["Current Ver", "Android Ver"], axis=1, inplace=True)

In [ ]:
playstore['Rating'].fillna(playstore['Rating'].mean(), inplace=True)

/tmp/ipykernel_147670/306629046.py:1: FutureWarning:

A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.





In [ ]:
userreviews.dropna(inplace=True)

In [ ]:
playstore.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10356 entries, 0 to 10840
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10356 non-null  object 
 1   Category        10356 non-null  object 
 2   Rating          10356 non-null  float64
 3   Reviews         10356 non-null  float64
 4   Size            10356 non-null  float64
 5   Installs        10356 non-null  float64
 6   Type            10356 non-null  object 
 7   Price           10356 non-null  float64
 8   Content Rating  10356 non-null  object 
 9   Genres          10356 non-null  object 
 10  Last Updated    10356 non-null  object 
dtypes: float64(5), object(6)
memory usage: 970.9+ KB


In [ ]:
playstore.isnull().sum()

,0
App,0
Category,0
Rating,0
Reviews,0
Size,0
Installs,0
Type,0
Price,0
Content Rating,0
Genres,0


In [ ]:
userreviews.info()

<class 'pandas.core.frame.DataFrame'>
Index: 29692 entries, 0 to 64230
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     29692 non-null  object 
 1   Translated_Review       29692 non-null  object 
 2   Sentiment               29692 non-null  object 
 3   Sentiment_Polarity      29692 non-null  float64
 4   Sentiment_Subjectivity  29692 non-null  float64
dtypes: float64(2), object(3)
memory usage: 1.4+ MB


In [ ]:
userreviews.isnull().sum()

,0
App,0
Translated_Review,0
Sentiment,0
Sentiment_Polarity,0
Sentiment_Subjectivity,0


## Final Cleaning Summary

- Removed duplicate records from both datasets
- Handed all missing values (no null)
- Transforming data to relevant data type:
  - Integer : Prize, Installs, Size, Reviews
- Application Size is standarized to MB
- Coverted USD to INR for better interpretation
- Dropped irrelevant columns

### Final Result:
Clean, structured, and analysis-ready dataset with zero missing values

## Finding Insights, Correlation & Trends

### Finding Correlation

In [ ]:
playstore.corr(numeric_only=True)

,Rating,Reviews,Size,Installs,Price
Rating,1.000000,0.068604,0.063019,0.050755,-0.020600
Reviews,0.068604,1.000000,0.107815,0.634996,-0.009418
Size,0.063019,0.107815,1.000000,0.050116,-0.020539
Installs,0.050755,0.634996,0.050116,1.000000,-0.011148
Price,-0.020600,-0.009418,-0.020539,-0.011148,1.000000


In [ ]:
px.imshow(
    playstore.corr(numeric_only=True),
    text_auto=True,
    title="Correlation Matrix"
).show()

- Review and Installs have strong positive correlation (~0.63) which means more popular apps have more reviews
- Rating has weak correaltion with Installs and review
- Price has negative correlation with Rating, Installs, Reviews which means user prefers free application more over the paids one.

### Category vs Installs

In [ ]:
CategoryInstalls = playstore.groupby("Category")['Installs'].mean().sort_values(ascending=False)

px.bar(
    x = CategoryInstalls.index,
    y = CategoryInstalls.values,
    title="Category vs Installs",
).show()

- Communication category has the highest average installs, followed by Social and Video Players.
- Categories such as Medical, Events, and Beauty have very low installs.

### Free vs Paid Apps (Box Plot)

In [ ]:
px.box(
    playstore,
    x="Type",
    y="Installs",
    title="Category vs Installs",
).show()

- Free apps dominate in terms of installs, with significantly higher median and maximum values.
- Paid apps has lower install counts, showing limited reach.
- This suggests that users strongly prefer free applications.

### Rating Distribution

In [ ]:
appDistFig = px.histogram(
    playstore,
    nbins=20,
    x="Rating",
    title="Distribution of App Ratings",

)
appDistFig.update_layout(bargap=0.1)
appDistFig.show()

- Most apps are rated between 4.0 and 4.5 and very few with 5.
- Very few apps have ratings below 3.0.
- This indicates that the majority of apps maintain a good user experience.

### Size vs Installs

In [ ]:
px.scatter(
    playstore,
    x="Size",
    y="Installs",
    log_y=True,
    title="App Size vs Installs",
    opacity=0.5,
    color="Type"
).show()

- A high concentration of apps is observed in the lower size range (0–20 MB).
- This indicates that most apps on the Play Store are relatively small in size.

### Size Vs Reviews

In [ ]:
px.scatter(
    playstore,
    x="Size",
    y="Reviews",
    log_y=True,
    title="App Size vs Reviews",
    opacity=0.5,
    color="Type"
).show()

- Apps across all size ranges (small to large) achieve both low and high install counts.
- There is no consistent upward or downward trend between size and installs.
- Size is not a deciding factor for popularity

### Reviews vs Installs

In [ ]:
px.scatter(
    playstore,
    x="Reviews",
    y="Installs",
    trendline="ols",
    title="Reviews vs Installs",
    color="Type",
).show()

In [ ]:
px.scatter(
    playstore,
    x="Reviews",
    y="Installs",
    log_x=True,
    log_y=True,
    trendline="ols",
    title="Reviews vs Installs (Log Scale)",
    color="Type",
).show()

- There is a strong positive relationship between reviews and installs, as it was visible in Correalation Heatmap
- Free apps dominate across all ranges of installs compared to paid apps.

### Sentiment Distribution

In [ ]:
sentimentCounts = userreviews['Sentiment'].value_counts()
px.pie(
    values = sentimentCounts.values,
    names = sentimentCounts.index,
    title="Sentiment Analysis"
).show()

- Around 64% of reviews are positive, which indicates overall user satisfaction.
- About 21% are negative, shows areas of improvement.
- Around 14% are neutral, showing mixed experiences.

### Merging data based on apps

In [ ]:
mergeDF = pd.merge(playstore, userreviews, on="App")
mergeDF.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Translated_Review,Sentiment,Sentiment_Polarity,Sentiment_Subjectivity
0,Coloring book moana,ART_AND_DESIGN,3.9,967.0,14.0,500000.0,Free,0.0,Everyone,Art & Design;Pretend Play,"January 15, 2018",A kid's excessive ads. The types ads allowed a...,Negative,-0.250,1.000000
1,Coloring book moana,ART_AND_DESIGN,3.9,967.0,14.0,500000.0,Free,0.0,Everyone,Art & Design;Pretend Play,"January 15, 2018",It bad >:(,Negative,-0.725,0.833333
2,Coloring book moana,ART_AND_DESIGN,3.9,967.0,14.0,500000.0,Free,0.0,Everyone,Art & Design;Pretend Play,"January 15, 2018",like,Neutral,0.000,0.000000
3,Coloring book moana,ART_AND_DESIGN,3.9,967.0,14.0,500000.0,Free,0.0,Everyone,Art & Design;Pretend Play,"January 15, 2018",I love colors inspyering,Positive,0.500,0.600000
4,Coloring book moana,ART_AND_DESIGN,3.9,967.0,14.0,500000.0,Free,0.0,Everyone,Art & Design;Pretend Play,"January 15, 2018",I hate,Negative,-0.800,0.900000


In [ ]:
mergeDF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40414 entries, 0 to 40413
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     40414 non-null  object 
 1   Category                40414 non-null  object 
 2   Rating                  40414 non-null  float64
 3   Reviews                 40414 non-null  float64
 4   Size                    40414 non-null  float64
 5   Installs                40414 non-null  float64
 6   Type                    40414 non-null  object 
 7   Price                   40414 non-null  float64
 8   Content Rating          40414 non-null  object 
 9   Genres                  40414 non-null  object 
 10  Last Updated            40414 non-null  object 
 11  Translated_Review       40414 non-null  object 
 12  Sentiment               40414 non-null  object 
 13  Sentiment_Polarity      40414 non-null  float64
 14  Sentiment_Subjectivity  40414 non-null

### Sentiment vs Rating

In [ ]:
px.box(
    mergeDF,
    x="Sentiment",
    y="Rating",
    title="Sentiment Analysis on Rating"
).show()

- Apps with positive sentiment tend to have slightly higher ratings.
- Neutral and negative sentiments have lower median ratings.
- difference is not very large so ratings alone do not fully capture user opinion.

## What we have achieved till now

1. Which apps have most installs
- As we have verified from our analysis that:
  - Free apps shows higher installs than paid apps.
  - Paid apps are in lower install ranges.
- Conclusion:
  - The assumption that free apps have more installs.

2. Installs & Reviews relation
- From scatter plot and correlation it shows High reviews lead to high installs:
  - Strong positive correlation (~0.63).
  - Apps with more installs have more reviews.
- Conclusion:
  - App popularity increases User engagement.

3. Category performance
- From category vs installs it is verified that:
  - Communication, Social, Video Players dominate.
  - Medical and Events shows limited user reach.

4. Sentiment affects ratings
- From sentiment vs rating:
  - Positive sentiment shows slightly higher rating
  - Difference is not very large for Neutral, Positive and Negative
- Conclusion:
  - Sentiment influences ratings but not strongly.

5. Rating impact on installs
- From correlation:
  - Very weak relationship (~0.05)
- Conclusion:
  - Ratings do not significantly affect installs.

## Project Summary

- This project analyzed Google Play Store data along with user reviews to identify the key factors influencing app success. The analysis included data cleaning, handling missing values, removing duplicates, and converting data into appropriate formats for accurate analysis.
- Exploratory Data Analysis (EDA) was performed using multiple visualization techniques such as histograms, scatter plots, box plots, bar charts, pie charts, and heatmaps to uncover patterns, trends, and relationships in the data.
- Key aspects analyzed:
  - App installs, ratings, reviews, and pricing
  - Category-wise performance
  - Free vs Paid app behavior
  - User sentiment from reviews
  - Relationship between size, installs, and engagement

## Final Conclusion

- The analysis reveals that app success on the Google Play Store is primarily driven by user accessibility, engagement, and category demand rather than technical attributes like app size or even ratings alone.
- Free apps significantly outperform paid apps in terms of installs, indicating that pricing plays a crucial role in user acquisition. Additionally, there is a strong positive relationship between installs and reviews, showing that popular apps naturally attract higher user engagement.
- Category analysis shows that Communication, Social, and Video-related apps dominate the market, while other categories like Medical and Events have limited reach. This highlights the importance of selecting the right category for maximizing app visibility and success.
- Furthermore, app size does not show a significant impact on installs or reviews, confirming that both small and large apps can succeed depending on their value proposition.

### Important Insights
- Focus on a free applications to maximize reach
- Target high-demand categories for better visibility
- Improve user experience to increase engagement and reviews
- Use user feedback (sentiment) for continuous improvement

### Conclusion
- Overall, successful apps are those that combine minimal or free cost, usability, and strong user engagement rather than just technical specifications.